storages/jsonl-storage/src/lib.rs

```diff
@@ -37,10 +39,19 @@ impl JsonlStorage {
     }
 
     fn fetch_schema(&self, table_name: &str) -> Result<Option<Schema>> {
-        if !self.data_path(table_name).exists() {
+        let jsonl_exists = self.jsonl_path(table_name).exists();
+        let json_exists = self.json_path(table_name).exists();
+
+        if !jsonl_exists && !json_exists {
             return Ok(None);
         };
 
+        if jsonl_exists && json_exists {
+            return Err(Error::StorageMsg(
+                JsonlStorageError::BothJsonlAndJsonExist(table_name.to_owned()).to_string(),
+            ));
+        }
+
```

Comment:
nit. it would become more simple by using `match`

```rust
match (json_path.exists(), jsonl_path.exists()) {
    (true, true) => {
        return Err(Error::StorageMsg(
            JsonlStorageError::BothJsonlAndJsonExist(table_name.to_owned()).to_string(),
        ));
    }
    (true, false) => remove_file(json_path).map_storage_err()?,
    (false, true) => remove_file(jsonl_path).map_storage_err()?,
    (false, false) => {}
};
```